src/common/meta/src/ddl/physical_table_metadata.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use api::v1::SemanticType;
+use store_api::metadata::ColumnMetadata;
+use table::metadata::RawTableInfo;
+
+/// Generate the new physical table info.
+pub(crate) fn build_new_physical_table_info(
+    mut raw_table_info: RawTableInfo,
+    physical_columns: &[ColumnMetadata],
+) -> RawTableInfo {
+    let existing_columns = raw_table_info
+        .meta
+        .schema
+        .column_schemas
+        .iter()
+        .map(|col| col.name.clone())
```

Comment:
nitpick: We can collect the string reference without cloning it.
```suggestion
        .map(|col| &col.name)
```